src/modules/Elsa.Studio.Workflows/Shared/Components/DiagramDesignerWrapper.razor.cs

```diff
@@ -347,6 +368,33 @@ private async Task OnActivityEmbeddedPortSelected(ActivityEmbeddedPortSelectedAr
         var portProvider = ActivityPortService.GetProvider(portProviderContext);
         var embeddedActivity = portProvider.ResolvePort(portName, portProviderContext);
 
+        if (embeddedActivity == null)
+        {
+            // Lazy load.
+            if (activityDescriptor.CustomProperties.ContainsKey("WorkflowDefinitionVersionId"))
+            {
+                await InvokeWithBlazorServiceContext(async () =>
+                {
+                    var parentNodeId = activity.GetNodeId();
+                    var subGraph = await WorkflowDefinitionService.FindSubgraphAsync(WorkflowDefinitionVersionId, parentNodeId);
+
+                    if (subGraph != null)
```

Comment:
Should we not add some kind of feedback to the user that we couldn't find (load) the workflow if this is null?

Reply:
Yes, in fact we should throw an exception because this would only happen if there's a serialization bug.